cubano-core/src/test/java/org/concordion/cubano/driver/action/ActionWaitTests.java

```diff
@@ -0,0 +1,67 @@
+package org.concordion.cubano.driver.action;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertThat;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class ActionWaitTests {
+
+
+    @Test
+    public void withTimeout() throws Exception {
+        Clock clock = Clock.systemDefaultZone();
+        Instant start = clock.instant();
+
+        try {
+            ActionWait wait = new ActionWait()
+                    .withTimeout(TimeUnit.SECONDS, 2)
+                    .withPollingIntervals(TimeUnit.MILLISECONDS, 0, 100, 500)
+                    .withMessage("some data to appear");
+
+            wait.until(() -> {
+                return (String) null;
+            });
+
+            throw new IllegalStateException("Should have timed out");
```

Comment:
Will this ever time out? Consider using https://github.com/junit-team/junit4/wiki/timeout-for-tests.

Reply:
This is a catch incase someone changes something and the wait.until() doesn't time out.  This code should never be reached if the test and real code are working correctly.

Reply:
Code updated and more tests added

Reply:
So probably more that the timeout has finished without throwing a TimeoutException. 
I'd still suggest changing the test annotation to something like `@Test(timeout=3000)` in case it doesn't time out

Reply:
The code is now "fail("Test should never reach this point");", unfortunately this view doesn't reflect the code update.